bin/release.py

```diff
@@ -0,0 +1,125 @@
+
+# https://learn.microsoft.com/en-us/rest/api/azure/devops/release/releases/get-release?view=azure-devops-rest-6.0&tabs=HTTP#environmentstatus
+FAILED_STATUSES = ["canceled", "partiallySucceeded", "rejected"]
+COMPLETED_STATUSES = ["succeeded", "canceled", "partiallySucceeded", "rejected"]
+
+
+def create_ado_connection(organization, ADO_PAT) -> Connection:
+    """Returns an ADO connection to call the ADO Rest APIs."""
+
+    organization_url = "https://dev.azure.com/{0}".format(organization)
+    credentials = BasicAuthentication("", ADO_PAT)
+    connection = Connection(base_url=organization_url, creds=credentials)
+
+    return connection
+
+
+def get_release_definition(project, pipeline_name, release_client):
+    """Returns the ADO definition for the given pipeline"""
+    # Get all release definitions in a given project
+    project_release_definitions = release_client.get_release_definitions(project)
+
+    # Filter release definitions with the given pipeline name
+    pipeline_release_definitions = []
+    for definition in project_release_definitions.value:
+        if definition.name == pipeline_name:
+            pipeline_release_definitions.append(definition)
+
+    if pipeline_release_definitions is None or len(pipeline_release_definitions) == 0:
+        error_message = f"Pipeline named {pipeline_name} not found in project {project}"
+        raise Exception(error_message)
+
+    if len(pipeline_release_definitions) > 1:
+        error_message = (
+            f"More than 1 Pipeline named {pipeline_name} found in project {project}"
+        )
+        raise Exception(error_message)
+
+    return pipeline_release_definitions[0]
+
+
+def populate_release_metadata(project, pipeline_name, release_client):
+    """Returns release metadata with required information"""
+    release_definition = get_release_definition(project, pipeline_name, release_client)
+    release_metadata = {
+        "definitionId": release_definition.id,
+    }
+    return release_metadata
+
+
+def release_status_match(release, expected_status_list):
+    """Returns True if status of all the environments of the release match any of the expected statuses"""
+    # Each release can have one or more environments (stages).
+    release_env_statuses = [environment.status for environment in release.environments]
+    # return all(status in expected_status_list for status in release_env_statuses)
+    for status in release_env_statuses:
+        if status not in expected_status_list:
+            return False
+    return True
+
+
+def wait_for_release(release_client, project, release_id):
+    """Wait for the azure devops release to finish"""
+    release = release_client.get_release(project, release_id)
+
+    # polling interval is set in accordance with the rate limits specified here:
+    # https://learn.microsoft.com/en-us/azure/devops/integrate/concepts/rate-limits?view=azure-devops
+    polling_interval_secs = 30

```

Comment:
No reason we can't just use the full word here.
```suggestion
    polling_interval_seconds = 30

```